### ISSUE TITLE
FAPI compatibility

### ISSUE THREAD

**Issue by miltonbo** [GluuFederation/oxTrust#1924]
## Issue
I'm trying to use a FAPI env with oxTrust, but I'm facing an issue that I checked.
In order to use oxAuth with oxTrust to authenticate the user, I found a difference between FAPI authorize url and generated oxTrust authorize url.

For instance:
#### FAPI Authorization Url: 
https://ce-ob.gluu.org/oxauth/restv1/authorize?request=eyJraWQiOiJmYXBpLWNsaWVudCIsImFsZyI6IlBTMjU2In0.eyJhdWQiOiJodHRwczpcL1wvY2Utb2IuZ2x1dS5vcmciLCJzY29wZSI6Im9wZW5pZCIsImNsYWltcyI6eyJpZF90b2tlbiI6eyJhY3IiOnsidmFsdWUiOiJ1cm46bWFjZTppbmNvbW1vbjppYXA6c2lsdmVyIiwiZXNzZW50aWFsIjp0cnVlfX19LCJpc3MiOiJmYXBpLWNsaWVudC0xIiwicmVzcG9uc2VfdHlwZSI6ImNvZGUgaWRfdG9rZW4iLCJyZWRpcmVjdF91cmkiOiJodHRwczpcL1wvd3d3LmNlcnRpZmljYXRpb24ub3BlbmlkLm5ldFwvdGVzdFwvYVwvZ2x1dS10ZXN0XC9jYWxsYmFjayIsInN0YXRlIjoia25Ma1h2VlpVYyIsImV4cCI6MTU4MTY5ODAxNywibm9uY2UiOiJhYUxGY1hLQU1pIiwiY2xpZW50X2lkIjoiZmFwaS1jbGllbnQtMSJ9.JUts1TmfPnpnhKUQAH4tklchLF6bL90jOo5C1XkerjJKsgjvaATqFujD9iJpFYliM7NTGPLRaL-ZmgVLxdgrSjauNpveT3LW25s7eMue_t5xQko4z1EjSN1LSID_NAvyYiZ7z4qBmNHOQikr31PWqJ1vzIwM6P_YDzwk9K41hxIF_BGPZQLiLm0gvUmgvJtyw90QZgYYHnFYN2BjsC6Q84TehXOgkBnDmZISPPPfWSJUMas9dA2MvBkj4NHeMpkate3993Ipu7bH1Ksf8kKuZO-TF-Rs8AHaqwth4Ts2WkoWfUxXKsOIMUA4E1QcdiL6xZ5qolDT_ZHix6-LwM90eg&client_id=fapi-client-1&redirect_uri=https://www.certification.openid.net/test/a/gluu-test/callback&scope=openid&response_type=code%20id_token

#### oxTrust Authorization Url:
https://ce-ob.gluu.org/oxauth/restv1/authorize?client_id=1001.970fbe26-05d1-4cd2-b8a6-a2c8136cef7d&redirect_uri=https%3A%2F%2Fce-ob.gluu.org%2Fidentity%2Fauthcode.htm&response_type=code&scope=openid+profile+email+user_name&nonce=65c494a4-05b5-4906-857f-312cde989716&state=090a27c8-4d76-4d0e-b979-559ad9a7b4e3&acr_values=simple_password_auth

In the `AuthorizeRestWebServiceImpl` class, we have a validation where we should have `request` param to generate the `jwtRequest` variable and after that it uses `fapiCompatibility` flag to check whether it's not null. Please check out this line:

> https://github.com/GluuFederation/oxAuth/blob/0bda545ff388a8de1c94e96fbb9ce7378c71daaf/Server/src/main/java/org/gluu/oxauth/authorize/ws/rs/AuthorizeRestWebServiceImpl.java#L303

To summarize, `fapiCompability` flag is used in the whole deployment and it is validating that RP includes `request` param with many params as a JWT, but oxTrust doesn't do it.

## Steps to reproduce
1. Add this certificate in your browser, empty password: https://drive.google.com/open?id=1GiOq5KRmwq9IJicnmUvu1_dSaK2Hb6eb
2. Go to: https://ce-ob.gluu.org/identity

**Comment by yuriyz** [comment]
@miltonbo you can read `fapiCompatibility` flag on oxTrust side and if it's `true` construct request object on oxtrust side.

**Comment by miltonbo** [comment]
Added oxTrust json properties to process **Fapi MTLS** variant:
```
1. oxAuthClientFapiJWKPrivateKeyModulus
2. oxAuthClientFapiJWKPrivateKeyPrivateExponent
3. oxAuthFapiClientKeystoreType
4. oxAuthFapiClientKeystorePath
5. oxAuthFapiClientKeystorePassword
```

Changes done in oxCore and oxAuth client module.
